var/spack/repos/builtin/packages/adios2/package.py

```diff
@@ -188,7 +188,8 @@ class Adios2(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("libpng@1.6:", when="+png")
     depends_on("zfp@0.5.1:0.5", when="+zfp")
     depends_on("sz@2.0.2.0:", when="+sz")
-    depends_on("mgard", when="+mgard")
+    depends_on("mgard@2022-11-18:", when="+mgard")
+    depends_on("mgard@2023-01-10:", when="@2.9: +mgard")
```

Comment:
Any version after this fails with ADIOS2 (runtime testing).

Reply:
Can you elaborate, @vicentebolea? Do you mean that mgard@2023-12-09 (the preferred version) does not work with adios2? That sounds like a bug with either MGARD or ADIOS2. Can you be more specific about what you were trying to do that failed?

Reply:
In the ADIOS2 CI every tag after `mgard@2023-12-09` resulted in CI testing errors that dramatically increase the precision and runtime crashes. Perhaps that was resolved, however, the version right after mgard@2023-12-09 did not work as expected.

Reply:
Sorry for being dense, @vicentebolea, but what CI are we talking about? Is this some spack CI run with `spack unit-test`? Is this something run on a GitHub runner that I cannot see? Is this something pulled from the adios2 code repo? How do I replicate the test?

Reply:
@kmorel / @vicentebolea:

@vicentebolea approved this PR now:
>> Considering that things seem to be working well with recent MGARD versions in Spack, are you good with these changes?
>  Sorry, I forgot to approve the PR. Looks good to me :)

To make this PR merge-able, we need to set this conversation to resolved.

As @vicentebolea approved, I think it should no longer block merging, correct?